base/ca/src/main/java/com/netscape/cms/profile/def/CAValidityDefault.java

```diff
@@ -394,16 +394,45 @@ public void populate(IRequest request, X509CertInfo info)
         date.add(rangeUnit, range);
 
         Date notAfter = date.getTime();
-        logger.debug("CAValidityDefault: not after: " + notAfter);
+        logger.debug(method + " not after: " + notAfter);
+
+        /*
+         * by default, bypassCAvalidity is false, meaning that if the
+         * notAfter date passes the CA's notAfter date, then it will be
+         * auto-adjusted to match that of the CA's notAfter date
+         *
+         * On the other hand, if bypassCAvalidity is set to true, then
+         * the notAfter date will be set according to the 'range' setting.
+         * If the notAfter date passes that of the CA's then it's up to
+         * the validity constraint set by the profile
+         */
+        boolean bypassCAvalidity = getConfigBoolean(CONFIG_BYPASS_CA_NOTAFTER);
+        logger.debug(method + " populate: bypassCAvalidity=" + bypassCAvalidity);
+        // not to exceed CA's expiration
+        Date caNotAfter = null;
+        try {
+            caNotAfter =
+                mCA.getSigningUnit().getCertImpl().getNotAfter();
+        } catch (Exception e) {
+            // in case of root ca
+        }
```

Comment:
I'd suggest we check for the root CA case explicitly instead of ignoring all exceptions because if there's a bug that triggers an exception the validity check might be bypassed unintentionally.

IIRC the earlier code failed due to an NPE somewhere, so maybe we could check for `null` like this:
```
if (mCA.getSigningUnit() == null) {
    // in case of root ca, bypass validity check
} else {
    caNotAfter = mCA.getSigningUnit().getCertImpl().getNotAfter();
}
```
Feel free to use a different way to check for root CA case.